### ISSUE TITLE
Various warnings in React 16.13.1

### ISSUE THREAD

**Issue by yunyu** [coinbase/rest-hooks#300]
**React version** 16.13.1

**Concurrent mode** no

**Describe the bug**
React throws warnings when useResource is used anywhere:

```
Warning: Cannot update a component (`CacheProvider`) while rendering a different component (`PartnerRewards`). To locate the bad setState() call inside `PartnerRewards`, follow the stack trace as described in https://fb.me/setstate-in-render
```

**Comment by ntucker** [comment]
Thanks for the report!

"To locate the bad setState() call inside `PartnerRewards`, "

React seems to be suggesting the error is in PartnerRewards. I might be able to assist if the code is shared, but it most certainly is an interaction so hard to say what this is without being able to repro.

**Comment by yunyu** [comment]
Here is a barebones PartnerRewards that triggers the warning:

```
class PartnerRewardResource extends CMSResource {
  static urlRoot = `landing-page-partner-rewards/${SORT_BY_RANK}`;

  readonly id!: number;
  readonly company!: string;
  readonly blurb!: string;
  readonly website!: string;
  readonly logo!: Logo;
  readonly rank!: number;

  pk() {
    return this.id.toString();
  }
}

const PartnerRewards: React.FC<{}> = () => {
  const isDesktop = useBreakpoints({ minWidth: "desktop" });

  const rewardsData = useResource(PartnerRewardResource.listShape(), {});

  return (
    <Box bg="gray4">{JSON.stringify(rewardsData)}</Box>
  );
};
```

`CMSResource` is just an extended copy of `SimpleResource` with a custom fetch - it doesn't touch React.

**Comment by ntucker** [comment]
Thanks, was able to repro, looking into it!

**Comment by yunyu** [comment]
Appreciate it!

**Comment by ntucker** [comment]
Seems to be the call to fetch, but it seems odd that this would be a problem. I left a comment to get clarity in the react github: https://github.com/facebook/react/issues/18178#issuecomment-601992546

**Comment by ntucker** [comment]
Appears to be fixed in 4.5.9
